contracts/Core.sol

```diff
@@ -92,11 +128,130 @@ contract Core is CoreInterface {
 		return coreChainIdRemote;
 	}
 
+	/// @dev public function openSTRemote
+	/// @return address coreOpenSTRemote
 	function openSTRemote()
 		public
 		view
 		returns (address /* OpenSTRemote */)
 	{
 		return coreOpenSTRemote;
 	}
+
+	/// @dev public function getLatestStateRootBlockHeight
+	/// @return uint256 latestStateRootBlockHeight
+	function getLatestStateRootBlockHeight()
+		public
+		view
+		returns (uint256 /* latest state root block height */)
+	{
+		return latestStateRootBlockHeight;
+	}
+
+	/// @dev public function getLatestStorageRootBlockHeight
+	/// @return uint256 latestStorageRootBlockHeight
+	function getLatestStorageRootBlockHeight()
+		public
+		view
+		returns (uint256 /* latest storage root block height */)
+	{
+		return latestStorageRootBlockHeight;
+	}
+
+	/**
+     * Commit new state root for a block height
+     *
+     */
+	/// @dev Commit new state root for a block height
+	/// @param _blockHeight block Number for which stateRoots mapping needs to update
+	/// @param _stateRoot state root of input block height
+	/// @return bytes32 stateRoot
+	function commitStateRoot(
+		uint256 _blockHeight,
+		bytes32 _stateRoot)
+		external
+		returns(bytes32 stateRoot)
+	{
+		// check if the caller is whitelisted worker
+		require(workers.isWorker(msg.sender), "Invalid worker address");
+		// State root should be valid
+		require(_stateRoot != bytes32(0), "Invalid state root");
+		// Input block height should be valid
+		require(_blockHeight > latestStateRootBlockHeight, "Given Block height is lower than latestBlockHeight.");
+
+		stateRoots[_blockHeight] = _stateRoot;
+		latestStateRootBlockHeight = _blockHeight;
+
+		emit StateRootCommitted(_blockHeight, _stateRoot);
+
+		return stateRoot;
+	}
+
+	/// @dev Verify account proof and update storageRoots mapping
```

Comment:
/// @dev Verify account proof of OpenSTRemote and commit storage root at given block height